### DISCUSSION TITLE
Does this only work on Raspberry Pi 4 or can it run on 3?

### DISCUSSION THREAD

**Discussion by sayhelloto** [theengs/gateway#168] [Q&A]
Tried to run Theengs on a Raspberry Pi 3 but got this error when launching. Is this the expected behavoir on non 4 devices? I managed to run my Switchbot with https://github.com/RoButton/switchbotpy so i thought I would be good to go with Theengs Gateway.

INFO:BLEGateway:Starting BLE scan
ERROR:BLEGateway:[Errno 111] Connection refused

**Comment by koenvervloesem** [comment]
Theengs Gateway runs just fine on a Raspberry Pi 3.

Error 111 is about the connection to the MQTT broker, probably an invalid username or password. Can you double-check whether the MQTT settings are correct? And otherwise, show the output of `python -m TheengsGateway.diagnose`?
